fw/tls.c

```diff
@@ -333,8 +308,10 @@ tfw_tls_encrypt(struct sock *sk, struct sk_buff *skb, unsigned int limit)
 		sgt.nents += r;
 		out_sgt.nents += r;
 
+		skb_tail->sk = sk;
```

Comment:
Why would set `->sk` and reset it later?

Reply:
No, because it will be set later in `__tcp_transmit_skb`

Reply:
Yeah, the code makes questions. Apparently we need to set and immediately unset `skb_tail->sk` due to the actions with the TCP send queue it `__extend_pgfrags()` - both the places look hacky. Probably we need a small wrapping function in tcp.c, which will call `ss_skb_expand_head_tail()` and release the skb head. Maybe other solution is required, but the current code looks like a technical debt in future.

Reply:
Fixed